src/target/riscv/riscv-013.c

```diff
@@ -3563,8 +3602,11 @@ static int write_memory_abstract(struct target *target, target_addr_t address,
 		}
 
 		/* Execute the command */
-		result = execute_abstract_command(target, command);
+		unsigned int abstractcs_err;
+		result = execute_abstract_command(target, command, &abstractcs_err);
 
+		/* TODO: we need to modify error handling here. We can make decisions only
+		 * if abstractcs_err indicates an exception */
```

Comment:
I believe we should check here for `cmderr==2 (unsupported)`, rather than `cmderr==3 (exception)`. Do you agree?

```suggestion
		/* TODO: we need to modify error handling here. We can make decisions only
		 * if cmderr indicates "unsupported" */
```

Reply:
There is an identical comment also in `read_memory_abstract()`.

Reply:
> I believe we should check here for cmderr==2 (unsupported), rather than cmderr==3 (exception). Do you agree?

 we should check for both, actually. The intention was to check that some error code is set . That is cmderr is not NONE (happens in case of timeouts)

Reply:
I have continued the discussion here: https://github.com/riscv/riscv-openocd/pull/964#discussion_r1422155725

Please let me know if you share the opinion or see it differently.